### PR TITLE
[ICDS Dashboard] UI for THR report 

### PR DESCRIPTION
**Pull request by dborowiecki**
Hi @calellowitz @Rohit25negi

We had to close previous request (https://github.com/dimagi/commcare-hq/pull/24831) because of merge issue. Maciej have implemented UI for THR reports according to requirements given in the ticket (https://dimagi-dev.atlassian.net/browse/ICDS-671). Now if user has access to all features they can choose 'Take Home Ratio (THR)' Program area, where 'View by' field is disabled and date starts from July 2019. He have also covered new functionalities with tests.

Need QA: no
Environment: ICDS
Locally tested: yes

Regards,
Damian

### CODE REVIEW

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 1]
```diff
@@ -470,7 +503,8 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
 
     vm.isVisible = function(level) {
         return level === 0 || (vm.selectedLocations[level - 1] && vm.selectedLocations[level - 1] !== 'all') &&
-            !(vm.isIncentiveReportSelected() && level > 2) && !(vm.isLadySupervisorSelected() && level > 2);
+            !(vm.isIncentiveReportSelected() && level > 2) && !(vm.isLadySupervisorSelected() && level > 2) &&
+            !(vm.isTakeHomeRationReportSelected() && level > 3);
```
Comment (calellowitz) [inline custom/icds_reports/static/js/directives/download/download.directive.js]: This is only available at sector and above?

**Review by calellowitz — COMMENTED** [review]
Couple questions but generally looks fine

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 2]
```diff
@@ -389,6 +413,15 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
             init();
             vm.selectedFormat = vm.formats[0].id;
         } else {
+            var date = new Date();
+            vm.selectedYear = date.getFullYear();
+            if (vm.isTakeHomeRationReportSelected()) {
+                vm.selectedLevel = 5;
```
[the remark this answers is not in the document]
Comment (dborowiecki) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: I've checked and this solution was incorrect, selectedLevel for other reports didn't work properly. It should work fine now.

File: custom/icds_reports/static/js/directives/download/download.directive.js (hunk 1 above)
Comment (dborowiecki) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: Yes, only awc is excluded.

**Comment by lwyszomi** [comment]
@dborowiecki could you fix the tests

**Comment by dborowiecki** [comment]
@calellowitz  @lwyszomi 
I've fixed tests.

**Comment by Rohit25negi** [comment]
same question I asked  here(https://github.com/dimagi/commcare-hq/pull/24831)

1. have you tested this that it does not show when feature flag is off?
2. What are paremeter values sent based on different filters selected under this report?
3. Did you test it locally to pull download a report locally to test the integration with the Backend?

Also Please go through my other comments and see if those are implemented in this also

**Comment by dborowiecki** [comment]
@Rohit25negi 
1.Yes, feature flags were tested manually
2. Parameters:
Indicator level is 10,
Location id is based on selected location,
Aggregation level is based on location: 0 for national, 1 for state, etc. 
3. Yes it was locally tested

**Comment by Rohit25negi** [comment]
> @Rohit25negi
> 1.Yes, feature flags were tested manually
> 2. Parameters:
> Indicator level is 10,
> Location id is based on selected location,
> Aggregation level is based on location: 0 for national, 1 for state, etc.
> 3. Yes it was locally tested

Thanks, @dborowiecki , Once last questions, could you please confirm you tested the integration with backend by downloading the report with different filters.

**Comment by dborowiecki** [comment]
@Rohit25negi 
Yes, it was tested for different filters, including dates, locations and file formats.

**Review by Rohit25negi — APPROVED** [review]
Looks good to me. Can merge now. But wait until evening. @calellowitz  `[ICDS]` is fine? or we want something like `[ICDS dashboard]`. I think later is better. Would like to change before merging

File: custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js [hunk 3]
```diff
@@ -285,8 +287,65 @@ describe('Download Directive', function () {
...
+            var firstAvailableMonth = controller.months[0].id;
+            var actualFirstYear = controller.selectedYear;
+
+            assert.equal(expectedMonth, firstAvailableMonth);
+            assert.equal(expectedFirstYear, actualFirstYear);
+            clock.restore();
+        });
+
+        it('tests latest possible data choice on THR raport', function () {
+            var fakeDate = new Date(2023, 8, 15);
```
Comment (calellowitz) [inline custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js]: I am quite confused by this test, what is it doing?

**Review by calellowitz — APPROVED** [review]
One question on the tests but otherwise looks fine

**Comment by calellowitz** [comment]
Changed the name to include dashboard and merged. Would still like an answer on the question though

File: custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js (hunk 3 above)
Comment (dborowiecki) [reply custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js]: @calellowitz 
It checks if current month and year are available in dropbox. For example when current month is August  it assures that August is also possible selection.

**Pull request merged**